folium/plugins/minimap.py

```diff
@@ -0,0 +1,130 @@
+
+    toggle_display: bool, default False
+        Sets whether the minimap should have a button to minimise it.
+
+    auto_toggle_display: bool, default False
+        Sets whether the minimap should hide automatically
+        if the parent map bounds does not fit within the minimap bounds.
+        Especially useful when 'zoomLevelFixed' is set.
+
+    minimized: bool, default False
+        Sets whether the minimap should start in a minimized position.
+
+    Examples
+    --------
+    >>> MiniMap(tile_layer='Stamen WaterColor', position='bottomleft')
+    """
+
+    _template = Template("""
+        {% macro script(this, kwargs) %}
+
+        var {{ this.tile_layer.get_name() }} = L.tileLayer(
+        '{{ this.tile_layer.tiles }}',
+        {{ this.tile_layer.options }} );
+
+        var {{ this.get_name() }} = new L.Control.MiniMap( {{this.tile_layer.get_name()}},
+         {{ this.options }});
+        {{ this._parent.get_name() }}.addControl({{ this.get_name() }});
+
+        {% endmacro %}
+    """)  # noqa
+
+    def __init__(self, tile_layer=None, position='bottomright', width=150,
+                 height=150, collapsed_width=25, collapsed_height=25,
+
+                 zoom_level_offset=-5, zoom_level_fixed=None,
+                 center_fixed=False, zoom_animation=False,
+                 toggle_display=False, auto_toggle_display=False,
+                 minimized=False):
+
+        super(MiniMap, self).__init__()
+        self._name = 'MiniMap'
+
+        if tile_layer is None:
+            self.tile_layer = TileLayer()
+        elif isinstance(tile_layer, TileLayer):
+            self.tile_layer = tile_layer
+        else:
+            self.tile_layer = TileLayer(tile_layer)
+
+        options = {
+            'position': position,
+            'width': width,
+            'height': height,
+            'collapsedWidth': collapsed_width,
+            'collapsedHeight': collapsed_height,
+            'zoomLevelOffset': zoom_level_offset,
+            'zoomLevelFixed': zoom_level_fixed,
+            'centerFixed': center_fixed,
+            'zoomAnimation': zoom_animation,
+            'toggleDisplay': toggle_display,
+            'autoToggleDisplay': auto_toggle_display,
+            'minimized': minimized,
+        }
+        self.options = json.dumps(options, sort_keys=True, indent=2)
+
+    def render(self, **kwargs):
+        figure = self.get_root()
+        assert isinstance(figure, Figure), ('You cannot render this Element '
+                                            'if it is not in a Figure.')
+        super(MiniMap, self).render()
+
+        figure.header.add_child(JavascriptLink('https://cdnjs.cloudflare.com/ajax/libs/leaflet-minimap/3.6.1/Control.MiniMap.js'))  # noqa
+
+        figure.header.add_child(CssLink('https://cdnjs.cloudflare.com/ajax/libs/leaflet-minimap/3.6.1/Control.MiniMap.css'))  # noqa
+        figure.add_child(self.tile_layer)
```

Comment:
Is adding `self.tile_layer` to `figure` needed? It seems the tile layer is already in your template. Can you try if this line is indeed redundant?

Reply:
Good catch.  I'm still getting comfortable with the template  usage.  I've removed this line.